lib/mergeSchemas.js

```diff
@@ -0,0 +1,67 @@
+import { mergeSchemas as _mergeSchemas, introspectSchema, makeRemoteExecutableSchema } from "graphql-tools"
+import { ApolloLink } from "apollo-link"
+import { createHttpLink } from "apollo-link-http"
+import { setContext } from "apollo-link-context"
+import fetch from "node-fetch"
+
+import localSchema from "../schema"
+import { headers as requestIDHeaders } from "./requestIDs"
+
+function createConvectionLink() {
+  const httpLink = createHttpLink({
+    fetch,
+    uri: `${process.env.CONVECTION_API_BASE}/graphql`,
+  })
+
+  const middlewareLink = new ApolloLink((operation, forward) => forward(operation))
+
+  const authMiddleware = setContext((_request, context) => {
+    const locals = context.graphqlContext && context.graphqlContext.res.locals
+    const tokenLoader = locals && locals.dataLoaders.convectionTokenLoader
+    const headers = { ...(locals && requestIDHeaders(locals.requestIDs)) }
+    // If a token loader exists for Convection (i.e. this is an authenticated request), use that token to make
+    // authenticated requests to Convection.
+    if (tokenLoader) {
+      return tokenLoader().then(({ token }) => Object.assign(headers, { authorization: `Bearer ${token}` }))
+    }
+    // Otherwise use no authentication, which is also meant for fetching the service’s (public) schema.
+    return headers
+  })
+
+  return middlewareLink.concat(authMiddleware).concat(httpLink)
+}
+
+export async function mergeSchemas() {
+  const convectionLink = createConvectionLink()
+
+  const convectionSchema = await makeRemoteExecutableSchema({
+    schema: await introspectSchema(convectionLink),
+    link: convectionLink,
+  })
+
+  const linkTypeDefs = `
+    extend type Submission {
+      artist: Artist
+    }
+  `
+
+  return _mergeSchemas({
+    schemas: [localSchema, convectionSchema, linkTypeDefs],
+    // Prefer others over the local MP schema.
+    onTypeConflict: (_leftType, rightType) => {
+      console.warn(`[!] Type collision ${rightType}`) // eslint-disable-line no-console
+      return rightType
+    },
+    resolvers: mergeInfo => ({
+      Submission: {
```

Comment:
@alloy - can you explain how adding in additional schemas will work in relation to the above?

Reply:
No this is not a namespace, it’s the way in which schemas are created using `graphql-tools` https://github.com/apollographql/graphql-tools/blob/master/README.md. In this case the code you’re referring to is the implementation for the schema extension just above (in IDL).